components/claim/components/email_to_oauth.tsx

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useRef} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import classNames from 'classnames';
+
+import {AuthChangeResponse} from '@mattermost/types/users';
+
+import {emailToOAuth} from 'actions/admin_actions.jsx';
+
+import Constants, {ClaimErrors} from 'utils/constants';
+import {localizeMessage, toTitleCase} from 'utils/utils';
+import {t} from 'utils/i18n';
+
+import LoginMfa from 'components/login/login_mfa';
+import LocalizedInput from 'components/localized_input/localized_input';
+
+import {SubmitOptions} from './email_to_ldap';
+
+type Props = {
+    newType: string | null;
+    email: string;
+    siteName?: string;
+}
+
+const EmailToOAuth = (props: Props) => {
+    const [showMfa, setShowMfa] = useState(false);
+    const [password, setPassword] = useState('');
+    const [serverError, setServerError] = useState<string>('');
+    const passwordInput = useRef<HTMLInputElement>(null);
+
+    const preSubmit = (e: React.FormEvent) => {
+        e.preventDefault();
+
+        const password = passwordInput.current?.value;
+        if (!password) {
+            setServerError(localizeMessage('claim.email_to_oauth.pwdError', 'Please enter your password.'));
+            return;
+        }
+
+        setPassword(password);
+
+        setServerError('');
+
+        submit({loginId: props.email, password});
+    };
+
+    const submit = ({loginId, password, token = ''}: SubmitOptions) => {
+        emailToOAuth(
+            loginId,
+            password,
+            token,
+            props.newType,
+            (data: AuthChangeResponse) => {
+                if (data.follow_link) {
+                    window.location.href = data.follow_link;
+                }
+            },
+            (err: {server_error_id: string; message: string}) => {
+                if (!showMfa && err.server_error_id === ClaimErrors.MFA_VALIDATE_TOKEN_AUTHENTICATE) {
+                    setShowMfa(true);
+                } else {
+                    setServerError(err.message);
+                    setShowMfa(false);
+                }
+            },
+        );
+    };
+
+    const error = serverError ? <div className='form-group has-error'><label className='control-label'>{serverError}</label></div> : null;
```

Comment:
we could use the same component defined in the `email_to_ldap.tsx` file (maybe even create a separate file for that since we will probably use that in the other files as well) and move that directly to the return value

Reply:
Done. ErrorLabel is already a seperate file